### PR TITLE
feat: Storage Volume repository

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-596

All tests will pass in the final [PR](https://github.com/keboola/keboola-as-code/pull/1770).

[Large set of fixes and E2e tests](https://github.com/keboola/keboola-as-code/pull/1759) is divided into smaller PRs to do a review.

--------------

**Changes:**
- Added/refactor storage `Volume` repository.

-----------

Commend per commits.

image

### CODE REVIEW

File: internal/pkg/service/stream/storage/repository/volume/repo.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		cancel()
+		r.logger.Info(ctx, "closing volumes stream")
+		wg.Wait()
+		r.logger.Info(ctx, "closed volumes stream")
+	})
+
+	var errCh <-chan error
+	r.volumes, errCh = etcdop.SetupFullMirror(r.logger, r.schema.WriterVolumes().GetAllAndWatch(ctx, r.client)).StartMirroring(ctx, wg)
+
+	return <-errCh
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/repository/volume/repo.go:66]: Higher layers are easier if the `etcdop` layer is functional and well tested.

File: internal/pkg/service/stream/storage/repository/volume/repo.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+// AssignVolumes assigns volumes to a new file.
+func (r *Repository) AssignVolumes(cfg assignment.Config, fileOpenedAt time.Time) assignment.Assignment {
+	return assignment.VolumesFor(r.volumes.All(), cfg, fileOpenedAt.UnixNano())
+}
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/repository/volume/repo.go:72]: `Repository.AssignVolumes`  connets together well tested `assignment.VolumesFor`, and well tested `r.volumes` etcd mirror. 

E2E tests for the volumes assignment are part of the next file repository PR.

File: internal/pkg/service/stream/storage/repository/volume/repo_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		require.NoError(t, err)
+		assert.Equal(t, []volume.Metadata{volume1, volume2}, result)
+
+		result, err = r.ListReaderVolumes().Do(ctx).All()
+		require.NoError(t, err)
+		assert.Equal(t, []volume.Metadata{volume3, volume4}, result)
+	}
+	{
+		// Check etcd state
+		etcdhelper.AssertKVsString(t, client, `
```
Comment (Matovidlo) [inline internal/pkg/service/stream/storage/repository/volume/repo_test.go:109]: Why dont you use comparison with file as in previous PR?

**Review by Matovidlo — APPROVED** [review]
LGTM

File: internal/pkg/service/stream/storage/repository/volume/repo_test.go (hunk 3 above)
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/repository/volume/repo_test.go:109]: Thanks, typo, I will fix it.

**Review by hosekpeter — APPROVED** [review]
LGTM

**Pull request merged**